main.php

```diff
@@ -224,3 +231,29 @@ function tc_form() {
 	wp_nonce_field( 'themecheck-nonce' );
 	echo '</form>';
 }
+
+/**
+ * Used to allow some directories to be skipped during development.
+ *
+ * @param  string  $filename a filename/path
+ * @return boolean
+ */
+function tc_is_other_file_in_dev_directory( $filename ) {
+	$skip     = false;
+	// Filterable List of dirs that you may want to skip other files in during
+	// development.
+	$dev_dirs = apply_filters(
+		'tc_common_dev_directories',
+		array(
+			'node_modules',
+			'vendor',
```

Comment:
I wouldn't add `vendor` here by default, as it can be needed in production builds sometimes. `node_modules` can always be excluded.

Reply:
In many themes I've worked on the `node_modules` directory is also sent in production builds. `npm --save` vs `npm --save-dev`.

I should update the name of the filters to make it clearer what they do because they don't technically skip the directories even when the filters are used. The only cause it to skip files that are not `.php` or `.css` files. So even if /vendors/ dir is excluded via the filter it will still scan the .php files from there :)

In my testing the problem is not actually caused by scanning of importat files for the checks here - it's parsing of the binary file types in these directories which exhausts memory or hits timeout limits.

Reply:
Oh wow, node_modules should really never be in the production :S usually all js is just bundled and used in a project.
I can see the merit of completly skipping both folders entirely for theme check purposes (not to exhaust the memory)

Reply:
@dingo-d any chance you could be my 2nd set of eyes and test this with me? I have tested myself already with over a dozen different themes I had in my local but I always like to have an extra person look it over incase I missed anything.

Reply:
I've tested it with the company's boilerplate and it seems to be working. When I put `add_filter('tc_skip_development_directories', '__return_true')` in my themes `functions.php` the check works and I can see

```
WARNING: The theme is using `#wpadminbar` in node_modules/@wordpress/edit-post/build-style/style.css. Hiding the admin bar is not allowed. This warning must be manually checked.Line 56: body.js.is-fullscreen-mode #wpadminbar {
```

but without it, I get memory exhausted error on `theme-check/checks/nongplsites.php on line 11`.

When I added 

```php
add_filter( 'tc_common_dev_directories', function( $ignored_folders ) {
  $ignored_folders[] = '.storybook';

  return $ignored_folders;
} );
```

I could see that folder ignored from the 

```
REQUIRED: .phpstorm.meta.php .eslintrc .stylelintrc .gitignore .eslintignore .travis.yml Hidden Files or Folders found. This file must not be in a theme.
```

error. So all seems to be ok :+1: